wm/client.go

```diff
@@ -0,0 +1,236 @@
+// +build linux,!ci
+
+package wm
+
+import (
+	"github.com/BurntSushi/xgb/xproto"
+	"github.com/BurntSushi/xgbutil/ewmh"
+	"github.com/BurntSushi/xgbutil/icccm"
+	"github.com/BurntSushi/xgbutil/xevent"
+	"github.com/BurntSushi/xgbutil/xprop"
+
+	"fyne.io/desktop"
+	"fyne.io/fyne"
+)
+
+type clientMessageStateAction int
+
+const (
+	clientMessageStateActionRemove clientMessageStateAction = 0
+	clientMessageStateActionAdd    clientMessageStateAction = 1
+	clientMessageStateActionToggle clientMessageStateAction = 2
+)
+
+type client struct {
+	id, win xproto.Window
+
+	iconic    bool
+	maximized bool
+
+	allowedActions []string
+	stateHints     []string
+
+	f  *frame
```

Comment:
This is accessed a lot from other code - it would be much clearer as "frame" - the duplication sucks but it is more readable.